api/controller.go

```diff
@@ -2738,58 +2735,51 @@ func (c *Controller) ActionsListRunHooksHandler() actionsop.ListRunHooksHandler
 			return handleErr(err, catalog.ErrRepositoryNotFound)
 		}
 
-		before := time.Now()
-		if params.Before != nil {
-			before = time.Time(*params.Before)
-		}
-		hooksIter, err := c.deps.Actions.ListHooks(repo.Name, params.RunID, before)
+		after := swag.StringValue(params.After)
+		tasksIter, err := c.deps.Actions.ListRunTasks(deps.ctx, repo.Name, params.RunID, after)
 		if err != nil {
 			return handleErr(err, actions.ErrNotFound)
 		}
-		defer hooksIter.Close()
-
-		res := &actionsop.ListRunHooksOKBody{Pagination: &models.Pagination{HasMore: swag.Bool(true)}}
-		for hooksIter.Next() {
-			val := hooksIter.Value()
-			status := models.HookRunStatusCompleted
-			if !val.Passed {
-				status = models.HookRunStatusFailed
-			}
+		defer tasksIter.Close()
 
-			res.Results = append(res.Results, &models.HookRun{
+		payload := &actionsop.ListRunHooksOKBody{
+			Results: make([]*models.HookRun, 0),
```

Comment:
```suggestion
			Results: make([]*models.HookRun),
```

Reply:
This is on purpose so the caller doesn't need to handle nil - will get empty array